mautic/mautic
# Command for remove webhook log

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features, enhancements and bug fixes (until 3.3.0 is released) 
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:
Every time when webhook log is created then we check that logs limit is exceed or not as per defined limit in configuration. if limit exceed then we should delete log. now we have created a command for remove logs which are exceed max limit.
and added a configuration clean_webhook_logs_in_background if it is false then it will work same as earlier else log will be removed from command using cron job in background.


#### Steps to test this PR:
1. Create a webhook
2. Everytime when webhook will trigger then log will be added in mautic_webhook_logs table
3. At time of add new log , it should work as :-
      if clean_webhook_logs_in_background = false , logs more than defined limit should be deleted instantly
      if clean_webhook_logs_in_background = true , logs more than defined limit should be deleted using command via cron job once in 24 hours.

**Note :- Command should be execute from cronjob**




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10405"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
